File: lib/typed-mind-python/tests/fixtures/services.py
from typing import Optional, List, Protocol
from .models import User, UserRole


class Repository(Protocol):
    def find_by_id(self, id: int) -> Optional[User]: ...
    def find_all(self) -> List[User]: ...


class UserService:
    def __init__(self, role: UserRole) -> None:
        self._role = role

    async def find_by_id(self, id: int) -> Optional[User]:
        return None

    def find_all(self) -> List[User]:
        return []

    @staticmethod
    def create_default() -> "UserService":
        return UserService(UserRole.USER)

File: lib/typed-mind-rust/tests/fixtures/src/services.rs
use crate::models::{User, UserRole};

pub trait Repository {
    fn find_by_id(&self, id: u64) -> Option<User>;
    fn find_all(&self) -> Vec<User>;
}

pub struct UserService {
    role: UserRole,
}

impl UserService {
    pub fn new() -> Self {
        Self { role: UserRole::User }
    }

    pub fn find_by_id(&self, id: u64) -> Option<User> {
        None
    }
}

impl Repository for UserService {
    fn find_by_id(&self, id: u64) -> Option<User> { None }
    fn find_all(&self) -> Vec<User> { vec![] }
}
